	* gio/gsocket.c: Fix build on Solaris 8 by providing needed macros
	defined on Solaris 10.

	* gio/libasyncns/asyncns.c: Fix build on Solaris 8 by providing
	include of inttypes.h if stdint.h is not defined.
	Submitted at: https://bugzilla.gnome.org/show_bug.cgi?id=610131
	FIXED in glib 2.23.4.

	* gio/Makefile.in: Disable build in subdirectory tests, which does
	not build out of the box on Solaris

--- gio/gsocket.c.old	2010-01-07 01:19:12.000000000 +0100
+++ gio/gsocket.c	2010-01-29 13:07:02.275043000 +0100
@@ -55,6 +55,16 @@
 
 #include "gioalias.h"
 
+#if defined (sun) && !defined (CMSG_SPACE)
+/* Amount of space + padding needed for a message of length l */
+#define CMSG_SPACE(l)                                                   \
+        ((unsigned int)_CMSG_HDR_ALIGN(sizeof (struct cmsghdr) + (l)))
+
+/* Value to be used in cmsg_len, does not include trailing padding */
+#define CMSG_LEN(l)                                                     \
+        ((unsigned int)_CMSG_DATA_ALIGN(sizeof (struct cmsghdr)) + (l))
+#endif
+
 /**
  * SECTION:gsocket
  * @short_description: Low-level socket object
*** gio/libasyncns/asyncns.c.original   Mon Feb 15 16:20:38 2010
--- gio/libasyncns/asyncns.c    Mon Feb 15 16:24:36 2010
***************
*** 50,56 ****
--- 50,62 ----
  #endif

  #include <sys/resource.h>
+
+ #if HAVE_STDINT_H
  #include <stdint.h>
+ #endif
+ #if HAVE_INTTYPES_H
+ #include <inttypes.h>
+ #endif

  #ifdef HAVE_SYS_PRCTL_H
  #include <sys/prctl.h>
*** gio/Makefile.in.original	Tue Feb 16 17:28:32 2010
--- gio/Makefile.in	Tue Feb 16 17:28:56 2010
*************** ETAGS = etags
*** 258,264 ****
  CTAGS = ctags
  am__tty_colors = \
  red=; grn=; lgn=; blu=; std=
! DIST_SUBDIRS = libasyncns xdgmime inotify fen win32 . fam tests
  DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
  am__relativize = \
    dir0=`pwd`; \
--- 258,264 ----
  CTAGS = ctags
  am__tty_colors = \
  red=; grn=; lgn=; blu=; std=
! DIST_SUBDIRS = libasyncns xdgmime inotify fen win32 . fam 
  DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
  am__relativize = \
    dir0=`pwd`; \
*************** EXTRA_DIST = gio-marshal.list gio.symbol
*** 508,514 ****
  TEST_PROGS = 
  NULL = 
  SUBDIRS = $(am__append_1) $(am__append_2) $(am__append_6) \
! 	$(am__append_9) . $(am__append_12) tests
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@noinst_DATA = gio-2.0.lib
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@install_ms_lib_cmd = $(INSTALL) gio-2.0.lib $(DESTDIR)$(libdir)
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@uninstall_ms_lib_cmd = -rm $(DESTDIR)$(libdir)/gio-2.0.lib
--- 508,514 ----
  TEST_PROGS = 
  NULL = 
  SUBDIRS = $(am__append_1) $(am__append_2) $(am__append_6) \
! 	$(am__append_9) . $(am__append_12) 
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@noinst_DATA = gio-2.0.lib
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@install_ms_lib_cmd = $(INSTALL) gio-2.0.lib $(DESTDIR)$(libdir)
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@uninstall_ms_lib_cmd = -rm $(DESTDIR)$(libdir)/gio-2.0.lib
